### DISCUSSION TITLE
Rhombus Mission Statement, and Intended Audience

### DISCUSSION THREAD

**Discussion by DexterLagan** [racket/rhombus#193] [General]
Hi folks,

  I appologize in advance if this has been answered elsewhere. I have been reading everything Rhombus, and finally found this older page with its goals: https://github.com/racket/rhombus-brainstorming/blob/master/resources/goals.md

  I can't find any info regarding what the detailed design goals of the lang, and especially its target audience. From the perspective of a Racketeer, I'd be inclined to say 'Racket without parentheses', but that wouldn't be very helpful, and wouldn't even touch the question of user base. Here are a few random questions that spring to my mind:

- clearly there is an intention to reach a wider audience, but who exactly are we trying to reach? Python programmers? Lispers? First-time programmers?
- does the project include any kind of technical/performance improvements apart from a new syntax atop Racket?
- will there be a new, standalone interpreter that may bring said improvements if any? (I'm thinking... high-perf interpreter tuned for the Web? - I know that this really isn't the goal, but one can dream. PHP's success speaks volume about the need for a simple interpreted language that anybody can learn. Is Rhombus pretending to enter this arena? If not, why not?).
- are there considerations to keep in mind when choosing an editor/linter/plugins etc? I understand that DrRacket isn't going anywhere, but I can hardly see a new language coming out into the world without its preferred editor...
- is there a broader conversation going on towards reducing cognitive load? For example, basing oneself on recent research on programming and the brain, one could list characteristics that could be incorporated in the lang specs at some point.

  I'd like to thank all the people involved, and in particular Matthew Flatt and Jay McCarthy, for their initiative, immence effort, and bringing us the rare opportunity to contribute to the birth of a new language.

Cheers,

Dexter

**Comment by otherjoel** [comment]
As far as I have been able to tell, the whole value proposition is about the possibilities of hygienic macros/LOP atop a non-S-expression syntax. The intended audience is anyone who might find that idea useful and interesting. The size of that group might be unknown if nothing like this exists already. All the other points are kind of incidental.

**Comment by mflatt** [comment]
The "State of Rhombus" document here describes the current goals: https://docs.google.com/document/d/10GTdmxo6Uty_-SQY8hrz5unCwtNi_YIsuI5yghmZ6hU/edit#heading=h.uj866cbxcxds

A couple of additional answers:

* interpreter and performabce: Rhombus is intended to run on Racket, so a new interpreter or performance changes are not part of the current goal/plan.

* editors: The current Rhombus prototype includes DrRacket support for syntax coloring and indentation. We're updating the way those tools are built generally for Racket so they can be more easily used for other editors, like Emacs or through LSP.
